### ISSUE TITLE
List component Granularity and Structure

### ISSUE THREAD

**Issue by amaalej** [Talend/react-talend-components#40]
@jmfrancois Thank you very much for your contribution on the List component, however I have a plenty of remarks about the **Granularity** and **Structure** in the component:

**A - The Granularity :**
The list component should be a list of items without a toolbar, neither a head:
- The existing List component has a toolbar by default, I think it should be optional to add a toolbar.
- The existing List component has also a head, which I think should also be optional.

So, I think there are 7 cases:
**1 - Default:**
1.1- Default simple List
1.2- Default simple List + head
1.3- Default simple List + head + Toolbar
**2 - Large:**
2.1- Large simple List 
2.2- Large simple List + Toolbar
**3 - Tile:**
3.1- Tile simple List 
3.2- Tile simple List + Toolbar

To solve the toolbar issue, I propose to add a new props 'displayLayout' (for example) and the code could be like that:
```
if (props.displayLayout === 'withToolbar') {
		return (
			<div className="tc-list">
				<Toolbar {...toolbar} />
				{content}
			</div>
		);
	}
	else {
		return (
			<div className="tc-list">
				{content}
			</div>
		);
	}
```

To solve the head issue, here we have to rethink to whole List component structure, I explain this below in the **Structure** part.

**B - The Structure :**
Using the `<table>` tag, adds more constraints to the List component display, so I think it would be more flexible if we use flex to display the list items and to avoid the colspan headaches.

What do you think?

**Comment by frassinier** [comment]
I'm tryin to understand why there is an issue with the toolbar, did you ever see list component without toolbar in any mockup?

**Comment by acateland** [comment]
We are buildings blocks for Talend components not for everyone to use, the target is to build with the utmost ease different talend application with the closest user experience possible, this have been extensively and openly discussed prior any realisation.

**Comment by jsomsanith** [comment]
I disagree with the header, it shouldn't be optional in default list mode. I don't see the point to display a list with values without the knowledge of what is displayed.
For the toolbar, it is simple, if you don't need it just make a PR with a new boolean props, but only when you need it.
